IdSSLOpenSSL.pas

```diff
@@ -3299,7 +3284,24 @@ procedure TIdSSLContext.InitContext(CtxMode: TIdSSLCtxMode);
       SSL_CTX_clear_options(fContext, SSL_OP_NO_TLSv1_2);
     end;
   end;
+  if IsOpenSSL_TLSv1_3_Available then begin
+    if not(sslvTLSv1_3 in SSLVersions) then begin
+      SSL_CTX_set_options(fContext, SSL_OP_NO_TLSv1_3);
+    end 
+	else if (fMethod = sslvSSLv23) then begin
+      SSL_CTX_clear_options(fContext, SSL_OP_NO_TLSv1_3);
+    end;
+  end;
 
+  if sslvTLSv1_3 in SSLVersions then

```

Comment:
This set of `if`s feels backwards. Let's say `SSLVersions` is set to `[sslvTLSv1..sslvTLSv1_3]` (not unreasonable given that many people use `[sslvTLSv1..sslvTLSv1_2]`), then this series of `if`s would set the acceptable proto range to 1.2-1.3 only. But if the `if`s are swapped so that `sslvTLSv1` is tested first and `sslvTLSv1_3` last, then the proto range would be set to 1.0-1.3 instead, as the user intended.